### PR TITLE
Wrap grdproject

### PR DESCRIPTION
**Pull request by willschlitzer**
This pull requests wraps the GMT module [`grdproject`](https://docs.generic-mapping-tools.org/6.2/grdproject.html).

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

**Comment by meghanrjones** [comment]
Thanks for wrapping this module!

What do you think about adding a value-based check to the tests (using `npt.assert_allclose()`, similar to grdgradient tests), in addition to the current metadata-based checks?

File: pygmt/tests/test_grdproject.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+def test_grdproject_file_out(grid):
+    """
+    grdproject with an outgrid set.
+    """
+    with GMTTempFile(suffix=".nc") as tmpfile:
+        result = grdproject(grid=grid, projection="M10c", outgrid=tmpfile.name)
+        assert result is None  # return value is None
+        assert os.path.exists(path=tmpfile.name)  # check that outgrid exists
+        result = grdinfo(tmpfile.name, per_column=True).strip().split()
```
Comment (seisman) [inline pygmt/tests/test_grdproject.py:29]: Need to check the `result`, right?

Comment (willschlitzer) [reply pygmt/tests/test_grdproject.py:29]: Yes 🤦‍♂️

**Comment by willschlitzer** [comment]
> Thanks for wrapping this module!
> 
> What do you think about adding a value-based check to the tests (using `npt.assert_allclose()`, similar to grdgradient tests), in addition to the current metadata-based checks?

I'll add them in!

File: pygmt/tests/test_grdproject.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        npt.assert_approx_equal(float(result[0]), 0)  # x min
+        npt.assert_approx_equal(float(result[1]), 10)  # x max
+        npt.assert_approx_equal(float(result[2]), -5.55111512313e-17)  # y min
+        npt.assert_approx_equal(float(result[3]), 9.94585661273)  # y max
+        npt.assert_approx_equal(float(result[4]), -5130.48193359)  # min
+        npt.assert_approx_equal(float(result[5]), -152.585281372)  # max
```
Comment (weiji14) [inline pygmt/tests/test_grdproject.py:36]: This test is going to break we refactor `grdinfo` to return non-string outputs (as mentioned in https://github.com/GenericMappingTools/pygmt/issues/593#issuecomment-895144705). I'm debating whether we should check the statistics by loading the output grid into `xarray`, something like so:

https://github.com/GenericMappingTools/pygmt/blob/3f4090e60c12ba4b40e8b757ef1f199a04b1ff1a/pygmt/tests/test_triangulate.py#L97-L99

File: pygmt/src/grdproject.py [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    rectangular grid. If ``inverse`` is ``True``, it will project a
+    rectangular coordinate system to a geographic system. To obtain the value
+    at each new node, its location is inversely projected back onto the input
+    grid after which a value is interpolated between the surrounding input
+    grid values. By default bi-cubic interpolation is used. Aliasing is
+    avoided by also forward projecting the input grid nodes. If two or more
+    nodes are projected onto the same new node, their average will dominate in
+    the calculation of the new node value. The ``region`` parameter can be
+    used to select a map region larger or smaller than that implied by the
+    extent of the grid file.
```
Comment (weiji14) [inline pygmt/src/grdproject.py:40]: Description is missing 3 sentences from https://github.com/GenericMappingTools/gmt/blame/6.2.0/doc/rst/source/grdproject.rst#L40-L44.

File: pygmt/src/grdproject.py [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+)
+
+
+@fmt_docstring
+@use_alias(
+    G="outgrid",
+    J="projection",
+    I="inverse",
+    R="region",
+    V="verbose",
```
Comment (weiji14) [inline pygmt/src/grdproject.py:23]:
```suggestion
    V="verbose",
    n="interpolation",
    r="registration",
```

File: pygmt/src/grdproject.py [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+        The file name of the input grid or the grid loaded as a DataArray.
+    outgrid : str or None
+        The name of the output netCDF file with extension .nc to store the grid
+        in.
+    inverse : bool
+        When set to ``True`` transforms grid from rectangular to
+        geographical [Default is False].
+    {J}
+    {R}
+    {V}
```
Comment (weiji14) [inline pygmt/src/grdproject.py:56]:
```suggestion
    {V}
    {n}
    {r}
```

**Pull request merged**